src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RC2.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public abstract class RC2 : SymmetricAlgorithm
+    {
+        protected RC2()
+            : base()
+        {
+            KeySize = 128;
+            BlockSize = 64;
+        }
+
+        public static RC2 Create()
+        {
+            throw new NotImplementedException(SR.WorkInProgress);
+        }
+
+        public override KeySizes[] LegalKeySizes
+        {
+            get
+            {
+                return s_legalKeySizes.CloneKeySizesArray();
+            }
+        }
+
+        public override KeySizes[] LegalBlockSizes
+        {
+            get
+            {
+                return s_legalBlockSizes.CloneKeySizesArray();
+            }
+        }
+
+        public override int KeySize
+        {
+            get
+            {
+                return base.KeySize;
+            }
+
+            set
+            {
+                if (value < _effectiveKeySize)
+                    throw new CryptographicException(SR.Cryptography_RC2_EKSKS);
+                base.KeySize = value;
+            }
+        }
+
+        public virtual int EffectiveKeySize
+        {
+            get
+            {
+                if (_effectiveKeySize == 0)
+                    return KeySize;
+                return _effectiveKeySize;
+            }
+
+            set
+            {
+                if (value > KeySize)
+                    throw new CryptographicException(SR.Cryptography_RC2_EKSKS);
+
+                if (value == 0)
+                {
+                    _effectiveKeySize = value;
+                }
+                else if (value < 40)
+                {
+                    throw new CryptographicException(SR.Cryptography_RC2_EKS40);
```

Comment:
It does seem redundant but the original desktop code did set aside this specific error string for the  “< 40” case so I figured it was safer not to tamper.